lambda/update/parse/changes.go

```diff
@@ -305,6 +308,60 @@ func (p *Parser) Each(fn func(int, *Parser) []shared.FieldError, required ...int
 	for _, idx := range indexes {
 		changes := indexedChanges[idx]
 		subParser := &Parser{root: p.root + "/" + strconv.Itoa(idx), changes: changes}
+		fn(strconv.Itoa(idx), subParser)
+		p.errors = append(p.errors, subParser.errors...)
+	}
+
+	return p
+}
+
+// EachKey will run fn with a [Parser] for any indexed or uid keys.
+//
+// Consider the changes:
+//
+//	{"key": "/9ac5cb7c-fc75-40c7-8e53-059f36dbbe3d/thing", "old": null, "new": "a string"}
+//	{"key": "/9ac5cb7c-fc75-40c7-8e53-059f36dbbe3d/thing", "old": null, "new": "another string"}
+//
+// Then to parse to a list of strings s do:
+//
+//	parser.Each(func(i int, p *Parser) {
+//		var v string
+//		p.Field("/thing", v)
+//		s = append(s, v)
+//		return p.Consumed()
+//	})
+func (p *Parser) EachKey(fn func(string, *Parser) []shared.FieldError) *Parser {
+	fieldChanges := make(map[string][]changeWithPosition, len(p.changes))
+
+	for _, change := range p.changes {
+		parts := strings.SplitN(change.Key, "/", 3)
+		if len(parts) != 3 || parts[0] != "" {
+			p.errors = append(p.errors, shared.FieldError{Source: change.Source("/key"), Detail: "require index or actor UID"})
+			continue
+		}
+
+		_, validKeyErr := strconv.Atoi(parts[1])
+		if validKeyErr != nil {
+			_, validKeyErr = uuid.Parse(parts[1])
+		}
```

Comment:
I think remove this and not worry about validating keys in here, I'd like to keep application concerns out of this package